chore(docs): add a note about openfga not evaluating the tupleset (#264)
openfga/openfga.dev: docs/content/configuration-language.mdx

This can be used to indicate that **viewers** of a **folders** are **viewers** of all **documents** in that **folder**.

:::caution
<details>
<summary>

>   Note that <ProductName format={ProductNameFormat.LongForm}/> does not evaluate the related object (the word after `from`, also called the tupleset), it will only consider directly related objects, and will not perform any evaluation (e.g. any rewrites on that relation, or `*`, etc..).

Reviewer: ```suggestion
  Note that <ProductName format={ProductNameFormat.LongForm}/> does not evaluate the related object (the word after `from`, also called the tupleset). Instead, it will only consider directly related objects and will not perform any evaluation (e.g. any rewrites on that relation, or `*`, etc..).
```

I am still a bit unsure about using the term "directly related object" as it implies direct relationship.  We explained that direct relationship includes * (https://openfga.dev/docs/configuration-language#the-direct-relationship-keyword).  Maybe use the term directly assigned objects?

Reviewer 2: ```suggestion
  Note that <ProductName format={ProductNameFormat.LongForm}/>, when searching tuples that are related to the object (the word after `from`, also called the tupleset), it will only consider concrete objects that were directly assigned, and will throw an error if it encounters a`*` or a userset.
```

This can be used to indicate that **viewers** of a **folders** are **viewers** of all **documents** in that **folder**.

:::caution
Note that <ProductName format={ProductNameFormat.LongForm}/> does not allow the referenced relation (the word after `from`, also called the tupleset) to be referencing another relation or allow non-concrete types (type bound public access (`<object_type>:*`) or usersets (`<object_type>#<relation>`)) in its type restrictions and will throw a validation error when attempting to call `WriteAuthorizationModel`.

> Note: In the deprecated 1.0 schema version, the validation error will be thrown at the time of evaluation (`Check`, etc..).

Reviewer: Should we have link to the migration document so that the reader understands what is schema 1.0?

Author: Done

This can be used to indicate that **viewers** of a **folders** are **viewers** of all **documents** in that **folder**.

:::caution
<details>
<summary>
Note that when searching tuples that are related to the object (the word after `from`, also called the tupleset), <ProductName format={ProductNameFormat.LongForm}/> will not do any evaluation and will only consider concrete objects (of the form `object_type:object_id`) that were directly assigned, and will throw an error if it encounters any rewrites, or a `*`, a type bound public access (`object_type:*`) or a userset (`object_type:object_id#relation`).
</summary>

>   In the case below:

Reviewer: do we want to mention that the model below is 1.0.  In the case of model 1.1, typecheck restriction will prevent you from writing such a model?

  <AuthzModelSnippetViewer
  configuration={{
    type_definitions: [
      {
        type: 'user',
      },
      {
        type: 'folder',
        relations: {
          parent: {
            this: {},
          },
          ancestor: {
            computedUserset: {
              relation: 'parent',
            }
          },
          editor: {
            union: {
              child: [
                {
                  this: {},
                },
                {
                  tupleToUserset: {
                    tupleset: {
                      relation: 'ancestor',
                    },
                    computedUserset: {
                      relation: 'editor',
                    },
                  },
                },
              ],
            },
          },
        },
      },
    ],
  }}
/>

  If you add two tuples:
  <WriteRequestViewer
    skipSetup={true}
    relationshipTuples={[
    {
      user: 'folder:product',
      relation: 'parent',
      object: 'folder:planning'
    },
    {
      user: 'user:anne',
      relation: 'editor',
      object: 'folder:product'
    }
  ]} />

>   Run the following check will return `false` because the `ancestor` relation contains a rewrite which will not be evaluated.

Reviewer 2: ```suggestion
  Running the following check will return `false` because the `ancestor` relation contains a rewrite which will not be evaluated.
```
